keps/sig-cli/5295-kyaml/README.md

```diff
@@ -0,0 +1,1428 @@
+
+- [X] (R) Enhancement issue in release milestone, which links to KEP dir in [kubernetes/enhancements] (not the initial KEP PR)
+- [ ] (R) KEP approvers have approved the KEP status as `implementable`
+- [X] (R) Design details are appropriately documented
+- [ ] (R) Test plan is in place, giving consideration to SIG Architecture and SIG Testing input (including test refactors)
+  - [ ] e2e Tests for all Beta API Operations (endpoints)
+  - [ ] (R) Ensure GA e2e tests meet requirements for [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md)
+  - [ ] (R) Minimum Two Week Window for GA e2e tests to prove flake free
+- [X] (R) Graduation criteria is in place
+  - [ ] (R) [all GA Endpoints](https://github.com/kubernetes/community/pull/1806) must be hit by [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md)
+- [ ] (R) Production readiness review completed
+- [ ] (R) Production readiness review approved
+- [X] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+This KEP proposes to add a new output format for kubectl,
+called "KYAML"
+(as in `kubectl get -o kayml ...`).
+This format is a strict subset (aka "dialect") of standard YAML,
+and so should be parseable by the existing ecosystem.
+This dialect seeks to emphasize syntactical choices which avoid
+many of the most common traps in YAML.
+For example, unlike standard YAML output,
+this dialect is not whitespace-sensitive,
+which makes it vastly easier to patch correctly in things like Helm charts.
+
+This KEP further proposes to make KYAML the standard format
+for all project-owned documentation and examples.
+
+## Motivation
+
+The vast majority of examples of how to use Kubernetes include YAML.
+These all use the most "conventional" YAML syntax,
+which is what `kubectl get -o yaml` also uses.
+Simple YAML is easy to read, but complex YAML can become very tricky.
+Some of the things that motivated this KEP are:
+
+Whitespace being significant means that
+readers and writers must track the nesting depth.
+While this does not sound too onerous,
+We have ample evidence that it is, in fact, very hard.
+Wrongly indented YAML is syntactically valid but does not marshal properly.
+Perhaps worst of all,
+template-expanding systems,
+such as the widely used `helm`,
+force users to get whitespace "just right"
```

Comment:
Helm charts done with kyaml would be much less readable/maintainable with all the curlies already required by Go templating.

Also while indentation isn't a parsing issue in kyaml, it certainly is an issue wrt readability.
It's disingenuous to say that the `nindent`s will go away.
They won't since the output of `helm template ...` needs to be human readable/parsable.

Reply:
helm template | yamlfmt (or yq) works

Reply:
I would argue the development of Helm templates will be more difficult. Most users of Helm never look at the generated output. They just see what's installed.

My concern would be that YAML structures and practices for k8s (helm included) are already widely set. Making changes to that is going to cause confusion.

The developer experience for Helm templates, and k8s yaml based things in general, will go down by using a style that's not widely used or documented in the world at large.